src/ztgamemgr.rs

```diff
@@ -0,0 +1,180 @@
+// ztgamemgr module has functions to interact with the live zoo stats such as cash, num animals, species, guests, etc.
+
+use crate::add_to_command_register;
+use crate::debug_dll::{get_from_memory, get_string_from_memory};
+
+use tracing::info;
+use std::collections::HashMap;
+use std::fmt;
+use std::fmt::Display;
+use num_enum::FromPrimitive;
+
+const GLOBAL_ZTGAMEMGR_ADDRESS: u32 = 0x00638048;
+
+// ZTGameMgr struct
+#[derive(Debug)]
+#[repr(C)]
+struct ZTGameMgr {
+    instance: u32, // 0x0
+    get_date: String, // 0x1194 (8 bytes)
+    get_cash: f32, // 0x0C
+    set_cash: f32, // 0x0C
+    add_cash: f32, // 0x0C
+    sub_cash: f32, // 0x0C
+    num_animals: u32, // 0x30
+    num_species: u32, // 0x38
+    num_guests: u32, // 0x54
+    num_tired_guests: u32, // 0x3C
+    num_hungry_guests: u32, // 0x40
+    num_thirst_guests: u32, // 0x44
+    num_guests_restroom_need: u32, // 0x48
+    num_guests_in_filter: u32, // 0x54
+    zoo_admission_cost: f32, // 0x1160
+    enable_dev_mode: bool, // 0x63858A
+}
+
+// SYSTEMTIME struct from Windows API
+#[derive(Debug)]
+#[repr(C)]
+struct SYSTEMTIME {
+    w_year: u16,
+    w_month: u16,
+    w_day_of_week: u16,
+    w_day: u16,
+    w_hour: u16,
+    w_minute: u16,
+    w_second: u16,
+    w_milliseconds: u16,
+}
+
+impl ZTGameMgr {
+    // returns the address of the ZTGameMgr instance in memory
+    fn instance() -> u32 {
+        get_from_memory::<u32>(GLOBAL_ZTGAMEMGR_ADDRESS)
```

Comment:
If the ZTGameMgr struct matches what's in memory you can just use `get_from_memory::<ZTGameMgr>(GLOBAL_ZTGAMEMGR_ADDRESS)`

Reply:
Ah okay. That should help with the previous concerns I think. I'll add padding alignment this week and refactor the struct.